src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8697,21 +8697,21 @@ private TypeWithAnnotations GetDeclaredParameterResult(ParameterSymbol parameter
 
         public override BoundNode? VisitIndexOrRangePatternIndexerAccess(BoundIndexOrRangePatternIndexerAccess node)
         {
-            BoundExpression receiver = node.Receiver;
-            var receiverType = VisitRvalueWithState(receiver).Type;
-            // https://github.com/dotnet/roslyn/issues/30598: Mark receiver as not null
-            // after indices have been visited, and only if the receiver has not changed.
-            _ = CheckPossibleNullReceiver(receiver);
-
             VisitRvalue(node.Argument);
```

Comment:
>VisitRvalue(node.Argument); [](http://example.com/codeflow?start=12&length=27)

The order of evaluation looks wrong here. It should be receiver, then argument, ...

Reply:
I don't know how to fix that. Feels like we need to keep the receiver as a placeholder for things (local rewriter and nullable walker) to compose well, but then that'd break CheckValue :-/
Let me think about it. I may file a follow-up issue as I'm not able to think straight at the moment.

Reply:
> Let me think about it. I may file a follow-up issue

When we visit the first argument placeholder (even if there are two of them), we can visit the argument as an rValue as well. I think this should work. Obviously, this need an additional state for the visitor

Reply:
Also, it feels like there is test gap, unless there is an existing failing test

Reply:
That works. Thanks